### ISSUE TITLE
Serialization

### ISSUE THREAD

**Issue by hitchhiker** [Blazored/LocalStorage#18]
As far as I can tell you're passing all values during 'Set' through JsonSerializer.ToString

This means it's impossible to use my own serializer as I can't change the serializer settings to use custom types, and I can't pre-serialize and send as a string without it being converted.

**Comment by chrissainty** [comment]
That’s correct, serialisation is handled automatically via the `JsonSerializer`. Which, as you’ve reasoned, means using a custom serialiser isn’t possible right now. 

Are you asking for this as a feature or just looking for confirmation?

**Comment by hitchhiker** [comment]
Just pointing it out. It's probably going to come up a lot, most people have requirements for specific serializer settings. Perhaps just expose the two serialization methods as Func<>, or allow us to swap in diff serialization providers etc

**Comment by chrissainty** [comment]
That’s cool just wasn’t sure from your opening post.

I agree, it would make sense to make these more customisable. I’ll have a look into making it a configurable option. Thanks for the feedback.

**Comment by benhysell** [comment]
I just ran into this using Telerik's Grid and saving / restoring Grid State: 

https://docs.telerik.com/blazor-ui/components/grid/state

tldr, they are currently depending on Newtonsoft...

```
// to store the serialized grid state, we need to have a custom serialized
// based on Newtonsoft.Json serialization. In the future this may not be required
public class FilterDescriptorJsonConverter : JsonConverter
{
    public override bool CanConvert(Type objectType)
    {
        return objectType == typeof(FilterDescriptorBase);
    }

    public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
    {
        JObject filterDescriptor = JObject.Load(reader);

        return filterDescriptor.ToObject<FilterDescriptor>(serializer);
    }

    public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
    {
    }
}
```

Not for nothing, and way off topic, I'm not diving head first into the new System.Text.Json.  I'm finding too many inconsistencies / unexpected behaviors to what I'm used to from Newtonsoft.  I'm willing to sacrifice the new 'speed' System.Text.Json brings for expected behavior.

**Comment by chrissainty** [comment]
Thanks for your comment @benhysell.

Right now I don't have any capacity to work on this feature. I'm happy to take a PR if you want to help out. Otherwise, the current setup will stay as is.

**Comment by benhysell** [comment]
@chrissainty Roger that!  Before diving in any pointers / concerns where this wouldn't be possible / current architecture would make this change difficult?

**Comment by benhysell** [comment]
And of course the reason I needed this is fixed in the next release of Telrik Blazor: 
https://feedback.telerik.com/blazor/1463789-save-load-grid-state-do-not-depend-on-custom-serialize-method

**Comment by chrissainty** [comment]
😆 Typical!

**Comment by musictopia2** [comment]
I saw the issue of serialization.  I have one suggestion which I can do.  I could create an interface for serialization with 2 implementations.  If that is done, then for the extension for registering the interfaces, can take an enum to choose which 2 to use.  One would use the json serialiazers and the other would use the new built in one.  What do you think of that idea.  I think I could do that part so users have a choice of what to use for serialization.

**Comment by IIARROWS** [comment]
If Blazored.LocalStorage just want to keep rely on System.Text.Json, it's sufficient to create a method **GetString** that returns the string exactly as found in storage and does not try to deserialize it.

The user can just deserialize it, if he wants. Or just use the string like it is (you know, sometime you just want to save a string).

**Comment by conficient** [comment]
I see that the `Task<String> GetStringAsync(string key);` option has been added but I cannot see this on the interfaces on the [current Nuget version - 2.1.6](https://www.nuget.org/packages/Blazored.LocalStorage/) at time of writing. That package is two months old. 

Is something broken on the publish pipeline?

**Comment by conficient** [comment]
Could it be this failed process? 
https://dev.azure.com/blazored/LocalStorage/_build/results?buildId=1018&view=results

**Comment by chrissainty** [comment]
This is now resolved with the merge of #88. Developers can now use this to serialise and deserialise any data they wish with their preferred JSON library.
